### DISCUSSION TITLE
PNPM version of Vercel is old

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [vercel/community#6865] [Help] [answered]
### Page to Investigate

https://github.com/FleetAdmiralJakob/chat-io/pull/136

### Steps to Reproduce

create a package.json with pnpm version 9.1.1 you will get:

ERR_PNPM_BAD_PM_VERSION  This project is configured to use v9.1.1 of pnpm. Your current pnpm is v9.0.4
![image](https://github.com/vercel/community/assets/77451351/27e6fc0b-3b33-4ec4-8998-cef2830c8d0e)

**Comment by swarnava** [comment, marked as answer]
Can you confirm if you are using `corepack` to use latest version of PNPM?

- set the environment variable `ENABLE_EXPERIMENTAL_COREPACK` to `1` following this guide: [vercel.com/docs/deployments/configure-a-build#corepack](https://vercel.com/docs/deployments/configure-a-build#corepack)
- set env var `COREPACK_ENABLE_STRICT` to `0` 
- set `package-manager-strict` to `false` in `.npmrc`

-------

Update: 3rd June, 2024

We've disabled the pnpm@9 strictness generally for Vercel builds. You will still see the pnpm warning in the build logs, but it should no longer fail the build.

If you are still seeing issues with this, please follow up https://github.com/vercel/vercel/issues/11607#issuecomment-2140817961.

**Reply by FleetAdmiralJakob** [reply]
the first step alone did not help.
together with the second step it worked!

**Comment by jthrilly** [comment]
The solution above is not actually a "solution" - it is merely a workaround.

The second and third steps (setting `COREPACK_ENABLE_STRICT` to 0, and `package-manager-strict` to `false`) simply disable strict version checking for the package manager, which allows the project to use Vercel's current default. This should not be needed.

This is being caused by PNPM >9 implementing strict enforcement, which is actually how corepack is designed to behave (see: https://github.com/pnpm/pnpm/issues/8087 and https://github.com/nodejs/corepack/issues/95).

-----

So back to the topic, I've set `ENABLE_EXPERIMENTAL_COREPACK` in my env in vercel, which should be enough. Can anyone at Vercel assist?

**Reply by MuellerMarvin** [reply]
I agree, I am having the same problem. Not sure why it can't just read it from the package.json

**Reply by fernandogprieto** [reply]
I have the same problem too

**Reply by alex-grover** [reply]
setting `ENABLE_EXPERIMENTAL_COREPACK=1` is working for me, Vercel uses pnpm@9.1.2 from my package.json.

**Reply by jthrilly** [reply]
If that's the case, they have fixed it (I had that variable set already). 

I opened a support case in my account earlier this week, so will update with whatever they reply with regarding what they did.

**Reply by k-xo** [reply]
@alex-grover do you have any other configurations like a `.npmrc` file or anything setup in your `package.json`? I've setup the variable but still failing when building on vercel

**Reply by jthrilly** [reply]
Vercel support just replied to me. Apparently they briefly rolled out 9.1.1 but reverted it due to what they called a "regression", which is actually https://github.com/pnpm/pnpm/issues/8087 ( so a "regression" is now a package implementing the correct behaviour inside a patch version - you can't make this stuff up! 😆).

So essentially, I suspect that they either disabled corepack for pnpm, or blacklisted that specific version when they found out that deploying it globally caused strict package manager checking to be enabled, likely as a result of this: https://github.com/vercel/vercel/issues/11607

**Reply by alex-grover** [reply]
builds still working for me 🤷🏻
<img width="943" alt="Screenshot 2024-05-22 at 09 05 18" src="https://github.com/vercel/community/assets/3088615/58447aee-60b9-461d-889d-c61b0179c621">

edit: I do not have pnpm listed in the package.json `engines` field

**Reply by IGassmann** [reply]
Removing `pnpm` from `engines` dit it for me.

**Comment by jthrilly** [comment]
Open issue here tracking this: https://github.com/vercel/vercel/issues/11607

**Comment by swarnava** [comment]
We've disabled the pnpm@9 strictness generally for Vercel builds. You will still see the pnpm warning in the build logs, but it should no longer fail the build.

If you are still seeing issues with this, please follow up https://github.com/vercel/vercel/issues/11607#issuecomment-2140817961.
